App/Redux/PhotoViewingRedux.ts

```diff
@@ -1,20 +1,36 @@
 import { createAction, ActionType, getType } from 'typesafe-actions'
 import Config from 'react-native-config'
 
-import { IFiles } from '@textile/react-native-sdk'
+import { IFiles, Thread } from '@textile/react-native-sdk'
+
+interface ThreadConfig {
+  name: string
+  type: Thread.Type
+  sharing: Thread.Sharing
+  whitelist?: ReadonlyArray<string>
+}
+
+interface ThreadOptions {
+  navigate?: boolean
+  selectToShare?: boolean
+  sharePhoto?: {
+    imageId: string
+    comment?: string
+  }
+}
 
 const actions = {
   insertThread: createAction('INSERT_THREAD', (resolve) => {
-    return (id: string, key: string, name: string) => resolve({ id, key, name })
+    return (id: string, key: string, name: string, type: Thread.Type, sharing: Thread.Sharing, whitelist: ReadonlyArray<string>) => resolve({ id, key, name, type, sharing, whitelist })
```

Comment:
Just curious why you're not using the `ThreadConfig` type defined above as a parameter in these actions (you're instead listing all the same properties as individual params). Maybe it's fine, just wondering.

Reply:
That's probably better-since some other parts of the codebase rely on these actions, and I would need to edit them as well, I decided to come back once I got the direct message feature working and refactor these. Fixing it now.